store/iavl/store.go

```diff
@@ -421,9 +421,13 @@ func (iter *iavlIterator) Value() []byte {
 	return val
 }
 
-// Implements types.Iterator.
+// Close closes the IAVL iterator by closing the quit channel and waiting for
+// the iterCh to finish/close.
 func (iter *iavlIterator) Close() {
 	close(iter.quitCh)
+	// wait iterCh to close
+	for range iter.iterCh {
```

Comment:
see comment in the original issue. I believe we can alternatively, wait on a done channel that gets sent during `iterateRoutine()`. But I think this will also suffice. In addition, not sure how to best test this as I imagine we'll need a large tree to have `iterateRoutine` still run after we call `Close()`.

Reply:
Use go test -race in unit test may make this bug happen. But it does not always happen.